instrumentation/sequel/lib/sequel/extensions/opentelemetry.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'sequel'
+
+# Sequel: The Database Toolkit for Ruby
+module Sequel
+  # The Sequel::OpenTelemetry extension
+  #
+  # @example Simple usage
+  #
+  #     require 'sequel-opentelemetry'
+  #     DB.extension :opentelemetry
+  #
+  # Related module: Sequel::OpenTelemetry
+  module OpenTelemetry
+    def tracer
+      ::OpenTelemetry::Instrumentation::Sequel::Instrumentation.instance.tracer
+    end
+
+    def trace_execute(super_method, sequel_method, sql, options, &block)
+      response = nil
+      attributes = {
+        ::OpenTelemetry::SemanticConventions::Trace::DB_NAME => opts[:database],
```

Comment:
`opts[:database]` is `nil` for jdbc connection like `jdbc:sqlite::memory:`. @jeremyevans is there any better way in here to get database name? I have checked `https://sequel.jeremyevans.net/rdoc/classes/Sequel/Database.html` and I wasn't able to find out.

Reply:
Sequel treats JDBC connection string URIs as opaque, it does not try to parse them.  You could try parsing the JDBC URI yourself if you want to find the database name.  Note that it isn't unusual to not have a database name even in the non-JDBC case.  `postgres:///?user=foo` will use the default database for user `foo` on PostgreSQL, but `opts[:database]` will still be nil in that case.

Reply:
I think the most robust way to handle this is to use `Database#database_type` to determine the type of database in use, then run database-specific code to get the database name using a query.  However, Sequel does not have built-in code to do that, and trying to handle it for all databases that Sequel supports seems challenging.

Considering that approach isn't really feasible, parsing the JDBC URI and returning the path part as the database name is probably the best you can do.  At least that gets you behavior similar to other adapters.

Reply:
I did some experiments, but it is not simple to extract the DB name from JDBC URI in unified way and also it is not possible to redact password in JDBC URI. Considering all this together, would it be acceptable to make this one optional @arielvalentin @kaylareopelle and call `compact` on attributes?

PS: It would be possible to add `DB_JDBC_DRIVER_CLASSNAME`, I can open additional PRO to enhance.

Reply:
It's a bit slower than checking if values are nil before adding them to the hash but using `compact!` is fine by me.

Reply:
Yes that's fine. Instrumentations should do their best to include attributes.